internal/model/unmarshal.go

```diff
@@ -50,3 +50,65 @@ func (t *tracesUnmarshaler) Unmarshal(buf []byte) (pdata.Traces, error) {
 	}
 	return td, nil
 }
+
+// MetricsUnmarshaler unmarshalls bytes into pdata.Metrics.
+type MetricsUnmarshaler interface {
```

Comment:
Is there a reason this is an interface? If someone needs to introduce a new marshaler, they can write one without having to satisfy this interface. I haven't seen any call sites that strictly depends on this interface contracts in the collector.

Ditto for the unmarshaler.

Reply:
This is internal, we are planning to use it and prove if we need it or not.